triedb/pathdb/difflayer.go

```diff
@@ -133,7 +221,32 @@ func (dl *diffLayer) node(owner common.Hash, path []byte, hash common.Hash, dept
 // Node implements the layer interface, retrieving the trie node blob with the
 // provided node information. No error will be returned if the node is not found.
 func (dl *diffLayer) Node(owner common.Hash, path []byte, hash common.Hash) ([]byte, error) {
-	return dl.node(owner, path, hash, 0)
+	if n := dl.cache.Get(hash); n != nil {
+		// The query from the hash map is fastpath,
+		// avoiding recursive query of 128 difflayers.
+		diffHashCacheHitMeter.Mark(1)
+		diffHashCacheReadMeter.Mark(int64(len(n.Blob)))
+		return n.Blob, nil
+	}
+	diffHashCacheMissMeter.Mark(1)
+
+	parent := dl.parent
```

Comment:
Try add a `origin` pointer to the disklayer in every difflayer such as it in snapshot's difflayer. 

Ref to https://github.com/bnb-chain/bsc/blob/master/core/state/snapshot/difflayer.go#L102

Reply:
in pathdb, generate new disklayer ,it is used for handling reorg, so difflayer can not hold disklayer pointer.

Reply:
Modify the `origin` pointer in the Cap().

Reply:
Updated.